Automattic/wp-calypso
# Checkout: Use CartMessages to display cart error messages in new checkout

#### Changes proposed in this Pull Request

Old checkout had a component called `CartMessages` which displays error messages returned by the `shopping-cart` endpoint. New checkout instead uses the hook `useDisplayErrors`. The important difference between these is that `CartMessages` replaces certain messages (those that contain inline HTML) with React versions.

This PR switches new checkout to use `CartMessages` instead. It preserves `useDisplayErrors` for loading errors, though.

As a side effect, this will also display any success messages from the `shopping-cart` endpoint (assuming there are no errors; I'm not sure why that's the case but it's the logic that was present in old checkout), which previously were being ignored in new checkout.

As another side effect, previously if there were multiple error messages, the notifications module would only display the last one. `CartMessages` combines messages so that the module will display all messages together.

Requires D48744-code to be merged first.

Fixes https://github.com/Automattic/wp-calypso/issues/45252

#### Screenshots

Before:

![Screen Shot 2020-08-27 at 6 55 44 PM](https://user-images.githubusercontent.com/2036909/91502816-fcaa9e80-e896-11ea-9902-4dc90b0898be.png)

After:

![Screen Shot 2020-08-27 at 6 26 04 PM](https://user-images.githubusercontent.com/2036909/91502550-55c60280-e896-11ea-857d-be8e74e5c662.png)


#### Testing instructions

- Apply D48744-code.
- Add a product to your cart, load checkout, and verify that there are no error messages displayed.
- Try to purchase when purchases are blocked (for testing this, you can apply D48747-code) and verify that you see an error message with a link that works, rather than raw HTML.
- Try to cause other error messages in the cart if you can and verify that they are displayed correctly.
